### PR TITLE
Multi measure models

### PR DESCRIPTION
**Pull request by dczulada**
This will still need more tests to be added.  However, since it will require more measure fixtures and test it probably makes more sense to wait until after the measure model updates are pulled in to do those.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/CYPRESS-343
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name: @okeefm 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

File: app/views/test_executions/_test_info.html.erb [hunk 1]
```diff
@@ -6,14 +6,20 @@
 
 <div class = 'col-sm-4'>
   <h1><%= info_title_for_product_test(task.product_test) %></h1>
-  <% unless task.product_test.is_a? ChecklistTest %>
+  <% if (task.product_test.is_a? MeasureTest) || (task.product_test.is_a? FilteringTest) %>
     <strong>Measure: </strong><span><%= task.product_test_name %></span><br/>
-  <% end %>
-  <strong>HQMF ID: </strong><span><%= task.measure_ids.join(',') %></span><br/>
```
Comment (lmd59) [inline app/views/test_executions/_test_info.html.erb]: It appears this line has been removed for the case where the test is a checklist test. Did we intend on removing the HQMF id from the "Record Sample Test Information" page?

Comment (dczulada) [reply app/views/test_executions/_test_info.html.erb]: That was unintentional.  Thou...I'm not sure how useful having the HQMF id on the Record Sample Test Execution. (below is a screenshot from the demo server with the HQMF ids)

image

File: app/models/multi_measure_cat3_task.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  def validators
+    @validators = [::Validators::MeasurePeriodValidator.new,
+                   ::Validators::QrdaCat3Validator.new(product_test.expected_results, true, true, false, product_test.bundle),
+                   ::Validators::CMSQRDA3SchematronValidator.new(product_test.bundle.version, false),
+                   ::Validators::ExpectedResultsValidator.new(product_test.expected_results)]
+    @validators
+  end
+
+  def execute(file, user)
+    te = test_executions.create!(expected_results: expected_results, artifact: Artifact.new(file: file), user_id: user)
```
[the remark this answers is not in the document]
Comment (okeefm) [reply app/models/multi_measure_cat3_task.rb]: I'm not sure we're changing anything on the user here that would require saving it; the way the `has_many`/`belongs_to` relation works in Rails is that the relation ID is kept on the element that belongs_to, in this case the Test Execution.

[the remark this answers is not in the document]
Comment (dczulada) [reply app/models/multi_measure_cat3_task.rb]: The user is associated with the test execution at the initialize step. This is slightly different than some of the other tasks, causing confusion

File: app/views/products/_cvu_plus_product.html.erb [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+<div class="product-test-tabs">
+
+<div class="product-test-tabs">
+  <ul>
+    <% each_tab(@product) do |test_type, title, _, html_id| %>
+      <li><a href = '<%= "##{html_id}" %>'><%= title %></a></li>
+    <% end %>
+  </ul>
+  <% each_tab(@product) do |test_type, title, description, html_id| %>
```
Comment (okeefm) [inline app/views/products/_cvu_plus_product.html.erb]: Isn't there only one tab in this scenario? Or are we anticipating adding additional test types as CVU+ matures?

**Pull request merged**